info-commands.go

```diff
@@ -246,7 +246,8 @@ type InfoMessage struct {
 	Versions     Versions           `json:"versions,omitempty"`
 	Usage        Usage              `json:"usage,omitempty"`
 	Services     Services           `json:"services,omitempty"`
-	Backend      interface{}        `json:"backend,omitempty"`
+	Backend      interface{}        `json:"backend,omitempty"` // Deprecated Apr 2023
+	ErasureInfo  ErasureBackend     `json:"erasureInfo,omitempty"`
```

Comment:
It seems like a hard-to-use deprecation.

Could you make it so `ErasureInfo` is always populated - even if an older server sends it as `Backend`?

Maybe instead of two fields do:
```suggestion
	Backend      ErasureBackend     `json:"backend"`
```
... and do a custom UnmarshalJSON on ErasureBackend, that handles the old types gracefully. That would simplify the deprecation and be backwards compatible both with "new mc -> old minio" and "old mc -> new minio".

(Also omitempty is pointless on structs)

Reply:
Actually you even need a custom UnmarshalJSON, since FSBackend will just be ErasureBackend with only "Type" set.